app/tasksmodule/datachecks.py

```diff
@@ -0,0 +1,178 @@
+        # TODO(nworden): figure out why setting task_retry_limit isn't working
+        retry_options = taskqueue.taskqueue.TaskRetryOptions(task_retry_limit=1)
+        taskqueue.add(name=name, method='POST', url=path,
+                      queue_name='datachecks', retry_options=retry_options,
+                      params={'cursor': cursor})
+
+    def alert(self, msg):
+        """Alerts developers of an error.
+
+        For now, we just raise an exception. It will show up in Stackdriver.
+        """
+        raise DatacheckException(msg)
+
+
+class PersonDataValidityCheckTask(DatachecksBaseTask):
+    """Checks that person records have valid data."""
+
+    BASE_NAME = 'person_data_validity_check'
+    TASK_PATH = 'check_person_data_validity'
+
+    def _check_person(self, person):
+        if not person.entry_date:
+            self.alert('A person record is missing an entry_date value.')
+        if not person.original_creation_date:
+            self.alert(
+                'A person record is missing an original_creation_date value.')
+        if (person.author_email and
+                not utils.validate_email(person.author_email)):
+            self.alert('A person record has an invalid author_email value.')
+
+    def post(self, request, *args, **kwargs):
+        del request, args, kwargs  # unused
+        q = model.Person.all(filter_expired=False).filter(
+            'repo =', self.env.repo)
+        cursor = self.params.get('cursor', '')
+        if cursor:
+            q.with_cursor(cursor)
+        try:
+            for person in q:
+                next_cursor = q.cursor()
+                self._check_person(person)
+                cursor = next_cursor
+        except runtime.DeadlineExceededError:
+            self._schedule_task(self.env.repo, cursor)
+        except datastore_errors.Timeout:
+            self._schedule_task(self.env.repo, cursor)
+        return django.http.HttpResponse('')
+
+
+class NoteDataValidityCheckTask(DatachecksBaseTask):
+    """Checks that notes have valid data."""
+
+    BASE_NAME = 'note_data_validity_check'
+    TASK_PATH = 'check_note_data_validity'
+
+    def _check_note(self, note):
+        if not note.entry_date:
+            self.alert('A note record is missing an entry_date value.')
+        if not note.original_creation_date:
+            self.alert(
+                'A note record is missing an original_creation_date value.')
+        if not note.person_record_id:
+            self.alert('A note record is missing a person_record_id value.')
+        if (note.author_email and not utils.validate_email(note.author_email)):
+            self.alert('A note record has an invalid author_email value.')
+        if (note.email_of_found_person and
+                not utils.validate_email(note.email_of_found_person)):
+            self.alert(
+                'A note record has an invalid email_of_found_person value.')
+        if not model.Person.get(self.env.repo, note.person_record_id):
```

Comment:
just checking, .get() isn't going to raise an exception while .filter(...).first() would return None?

Reply:
It'll return None if there's no result:
https://cloud.google.com/appengine/docs/standard/python/datastore/modelclass#Class_Methods